src/NHibernate/Loader/Loader.cs

```diff
@@ -599,7 +601,9 @@ private IEnumerable<SubselectFetch> CreateSubselects(IList<EntityKey[]> keys, Qu
 			}
 		}
 
-		internal void InitializeEntitiesAndCollections(IList hydratedObjects, object resultSetId, ISessionImplementor session, bool readOnly)
+		internal void InitializeEntitiesAndCollections(
+			IList hydratedObjects, object resultSetId, ISessionImplementor session, bool readOnly,
+			CacheBatcher cacheBatcher)
```

Comment:
For minimizing changes I have initially attempted to give a default `null` value to the added `cacheBatcher` parameter, but maca88/AsyncGenerator#115 forbids this by generating an invalid async counterpart in such case.

This change allows batching cache puts across queries in the case of query batch. Due to current `CacheBatcher` implementation, it will be limited to `batch-size` (according to entities mapping) and segregated by persister, which I do not consider very suitable, as discussed [here](https://github.com/nhibernate/nhibernate-core/pull/1633#discussion_r200837292).

Here too, for avoiding broadening the scope of the PR, I will leave this for a later improvement.

Reply:
>  maca88/AsyncGenerator#115 forbids this by generating an invalid async counterpart in such case

Fixed in version `0.8.2.9`.